Create tic-tac-commandline.py

MX-43 opened the pull request.

MX-43 committed:
Create tic-tac-commandline.py

siruma commented on tic-tac-commandline.py:
```diff
@@ -0,0 +1,206 @@
+        self.grid_slot6 = 2
+        self.grid_slot7 = 2
+        self.grid_slot8 = 2
+        self.grid_slot9 = 2
```
I suggest creating this as a single Python object that can be shared among all players.

MX-43 replied:
Added GameGrid class for storing the grid values

siruma commented on tic-tac-commandline.py:
```diff
@@ -0,0 +1,206 @@
+        print("game initiating")
+        self.active_player=1 #player 1 always starts
+        self.game_round=0    #game length max 9 rounds (=grid size)
+        self.game_finished=1 #game is inactive (1) when it's finished
```
Should this be a boolean value?

siruma commented on tic-tac-commandline.py:
```diff
@@ -0,0 +1,206 @@
+        self.grid_slot9 = 2
+    
+    def reset_game(self):
+        print("resetting game status...")
```
For improved server-side debugging, it is recommended to replace prints with logging.

MX-43 replied:
Replaced prints with logging

siruma commented on tic-tac-commandline.py:
```diff
@@ -0,0 +1,206 @@
+        
+
+    def forfeit_game(self, player_id):
+        print("forfeit by player ", player_id, "due to disconnection or lack of input...")
```
For improved server-side debugging, it is recommended to replace prints with logging.

siruma commented on tic-tac-commandline.py:
```diff
@@ -0,0 +1,206 @@
+        self.start_game()
+    
+    def start_game(self):
+        print("(re)start game...")
```
For improved server-side debugging, it is recommended to replace prints with logging.

siruma commented on tic-tac-commandline.py:
```diff
@@ -0,0 +1,206 @@
+        
+    def game_status(self):
+        print("this is the game status, to be delivered to connected clients")
+        print("contains: active_player, game_round, game_finished, grid_slots 1-9)")
```
For improved server-side debugging, it is recommended to replace prints with logging.

siruma commented on tic-tac-commandline.py:
```diff
@@ -0,0 +1,206 @@
+        print(self.grid_slot4, self.grid_slot5, self.grid_slot6)
+        print(self.grid_slot7, self.grid_slot8, self.grid_slot9)
+        print("#####")
+        print("#####")
```
For improved server-side debugging, it is recommended to replace prints with logging.

MX-43 replied:
All prints replaced with logging

siruma commented on tic-tac-commandline.py:
```diff
@@ -0,0 +1,206 @@
+        
+    def add_mark_to_slot(self, picked_slot):
+        print("picked slot:", picked_slot)
+        print("active_player", self.active_player)
```
For improved server-side debugging, it is recommended to replace prints with logging.

siruma commented on tic-tac-commandline.py:
```diff
@@ -0,0 +1,206 @@
+        if picked_slot==9 and self.grid_slot9==2 and self.active_player==0:
+            self.grid_slot9=0
+            self.active_player=1
+            self.game_round+=1
```
I think this would be more readable if the structure would be:
```
if self.active_player==1:
    if picked_slot==1 and self.grid_slot1==2:
       ...
elif self.active_player==0:
   if picked_slot==1 and self.grid_slot1==2:
      ....
else:
   # log the error
```

MX-43 replied:
Agreed, this makes it cleaner

siruma commented on tic-tac-commandline.py:
```diff
@@ -0,0 +1,206 @@
+        self.check_for_winner()
+            
+    def check_for_winner(self):
+        print("checking for winner...")
```
For improved server-side debugging, it is recommended to replace prints with logging.

siruma commented on tic-tac-commandline.py:
```diff
@@ -0,0 +1,206 @@
+        elif self.game_round==9:
+                self.declare_winner(0) #draw
+        else:
+            print("no winner yet...")
```
For improved server-side debugging, it is recommended to replace prints with logging.

siruma commented on tic-tac-commandline.py:
```diff
@@ -0,0 +1,206 @@
+    def declare_winner(self, player):
+        self.game_finished = 1
+        if player==0:
+            print("DRAW")
```
For improved server-side debugging, it is recommended to replace prints with logging.

siruma commented on tic-tac-commandline.py:
```diff
@@ -0,0 +1,206 @@
+        if player==0:
+            print("DRAW")
+        else:
+            print("WINNER IS PLAYER ", player)
```
For improved server-side debugging, it is recommended to replace prints with logging.

siruma commented on tic-tac-commandline.py:
Please move the file to the "src/server/" directory.

MX-43 committed:
cleaning up code

Print commands changed to logging commands. Fixed resetting bug.

MX-43 committed:
move tictac to src\server folder

MX-43 committed:
moved game grid to a separate class, clean code

moved game grid to a separate class GameGrid, cleaned code:
update class names, update grid to use list of values instead of separate variables, added function to display X/O/_ markers instead of values when printing game status to log

siruma reviewed (approved):
Looking good
